WordPress/gutenberg
# Fix classic block converted to regular blocks when clicking new 'Edit visually' button

## What?
Fixes #43209

When clicking the 'Edit visually' option on the block toolbar within the classic block, the block is converted to regular blocks.

This seems to be because the button didn't use the `toggleBlockMode` action like it should, instead it parsed the HTML into blocks and replaced the edited block.

I've also improved a few other issues:
- The button was being rendered in the `BlockSettingsMenu` component, it should be in the `BlockToolbar` component. To do this I had to move the component into a new folder.
- The button was doing `useSelect` to get the selected blocks, but it receives these as a prop.
- Removes an unnecessary spread of `...props`, the component only receives a single prop

One issue that I haven't tackled is that there are currently two 'Edit visually' buttons (one on the toolbar, one in the block settings menu), I'm treating that as out of scope.

## Testing Instructions
1. Insert a classic block
2. Edit it as HTML using the block settings dropdown
3. Toggle it back to 'Edit visually' using the toolbar button
4. The classic block should still be present.

## Screenshots or screencast 

### Before

https://user-images.githubusercontent.com/677833/184581013-ed7235e1-82e5-4aea-b24b-bfcdc204afa4.mp4


### After

https://user-images.githubusercontent.com/677833/184580867-ad7da918-a024-450f-af25-ba4e1399e987.mp4
